proteuslib/tests/test_user_scaling_with_chlorination.py

```diff
@@ -0,0 +1,838 @@
+
+        # NOTE: Give an initial guess of pH 7 and inlet values or 1e-6 for others
+        state_args = {'mole_frac_comp':
+                        {   'H2O': 1,
+                            'HOCl': 4e-6,
+                            'H_+': 10**-7/55.6,
+                            'NCl3': 1e-6,
+                            'NH2Cl': 1e-6,
+                            'NH3': 1e-6,
+                            'NH4_+': 1e-6,
+                            'NHCl2': 1e-6,
+                            'OCl_-': 1e-6,
+                            'OH_-': 10**-7/55.6
+                        },
+                        'pressure': 101325,
+                        'temperature': 300,
+                        'flow_mol': 10
+                    }
+        flags = fix_state_vars(model.fs.unit.control_volume.properties_out, state_args)
+        revert_state_vars(model.fs.unit.control_volume.properties_out, flags)
+
+        iscale.constraint_autoscale_large_jac(model)
+        jac, nlp = iscale.get_jacobian(model, scaled=True)
+        print("Extreme Jacobian entries:")
+        for i in iscale.extreme_jacobian_entries(jac=jac, nlp=nlp, large=100):
+            print(f"    {i[0]:.2e}, [{i[1]}, {i[2]}]")
+        print("Unscaled constraints:")
+        for c in iscale.unscaled_constraints_generator(model):
+            print(f"    {c}")
+        print("Scaled constraints by factor:")
+        for c, s in iscale.constraints_with_scale_factor_generator(model):
+            print(f"    {c}, {s}")
+        print("Badly scaled variables:")
+        for v, sv in iscale.badly_scaled_var_generator(model, large=1e2, small=1e-2, zero=1e-12):
+            print(f"    {v} -- {sv} -- {iscale.get_scaling_factor(v)}")
+        print(f"Jacobian Condition Number: {iscale.jacobian_cond(jac=jac):.2e}")
+
+    @pytest.mark.component
+    def test_initialize(self, chlorination_obj):
+        model = chlorination_obj
+
+        orig_fixed_vars = fixed_variables_set(model)
+        orig_act_consts = activated_constraints_set(model)
+
+        solver.options['bound_push'] = 1e-10
+        solver.options['mu_init'] = 1e-6
+        solver.options["nlp_scaling_method"] = "user-scaling"
+
+        state_args = {'mole_frac_comp':
+                        {   'H2O': 1,
+                            'HOCl': 4e-6,
+                            'H_+': 10**-7/55.6,
+                            'NCl3': 1e-6,
+                            'NH2Cl': 1e-6,
+                            'NH3': 1e-6,
+                            'NH4_+': 1e-6,
+                            'NHCl2': 1e-6,
+                            'OCl_-': 1e-6,
+                            'OH_-': 10**-7/55.6
+                        },
+                        'pressure': 101325,
+                        'temperature': 300,
+                        'flow_mol': 10
+                    }
+        model.fs.unit.initialize(state_args=state_args, optarg=solver.options, outlvl=idaeslog.DEBUG)
```

Comment:
To initialize, the same state_args as before must be passed to the initializer function. 

You must also set the 'bound_push' and 'mu_init' values as shown here, then set the scaling method to 'user-scaling'